### DISCUSSION TITLE
Suggestion: Make Copilot play nice with language-specific intellisense in VS Code

### DISCUSSION THREAD

**Discussion by jssblck** [community/community#65374] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Copilot is great in some contexts, but in others really gets in the way of intellisense provided by inline language servers. 

As an example, using Rust, if I create a struct definition and then later start typing an instantiation of the struct, `rust-analyzer` autocompletes available options. Similarly if I type `.` after a variable, I get an intellisense list of _known available_ methods to use.

However, if I have Copilot enabled, it prevents those autocompletes from appearing and (since it routinely hallucinates) the replacements are far worse.

In comparison, when I'm writing rustdoc example blocks I've found Copilot to be _extremely_ helpful. 

Currently, this leads to the unfortunate situation of needing to:
- Configure VS Code to disable auto inline completions (because they are so disruptive in code).
- Manually prompt for them when writing doc example blocks.

Ideally, I'd rather be able to configure "smart auto inline completions", where the criteria is:
- Show inline completions on a _blank line_ or a _comment only_ line.
- Show inline completions only on lines that match a user-provided regular expression.

**Comment by Akash1134** [comment]
Hi @jssblck ,

Thank you for sharing the feedback with such great examples and use cases. 

As a GitHub user, the most you can do is to share your feedback on this Discussion space with the correct tags (include examples), and trust that we are always working to make GitHub Copilot as awesome as possible. We can't guarantee that every bit of feedback sees action, but we are paying attention.

Thanks!
